### ISSUE TITLE
[BUG] login with member role, no secret api access

### ISSUE THREAD

**Issue by WuJun2016** [harvester/harvester#1894]
**Describe the bug**

The user login with the member role, views the vm template, and the page reports an error display no secret access

![image](https://user-images.githubusercontent.com/24985926/153567327-408f7717-5288-40b3-ba7c-b093de6e8bed.png)

![image](https://user-images.githubusercontent.com/24985926/153567416-4dd3a71f-dde7-46c4-8067-e303b4ccfc64.png)

The cloud-init in the template is saved via secret, so we need to have access to the secret
**To Reproduce**
1. login with member role
2. click  Advanced => Templates
3. click view config

**Expected behavior**

view config works fine

**Support bundle**

**Environment:**
 - Harvester ISO version: 
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630):

**Additional context**

---
# Dev update @johnliu55tw 
## Detail reproduce steps
1. Create a Harvester cluster (one node would be sufficient)
2. Follow the instructions on https://docs.harvesterhci.io/v1.0/rancher/rancher-integration/ to create a Rancher deployment and add the Harvester cluster into Rancher
3. In Rancher, create a new user `someuser` as "Standard User"
4. Go to the "Virtualization Management" page of Rancher -> Click "Manage" of the Harv. cluster -> Edit Config -> Add user `someuser` as "Member", instead of "Owner" or "Custom"
5. Logout from Rancher, and login with user `someuser`
6. Go to Virtualization Management -> Click the Harv. cluster -> Go to Advanced/Templates -> Click any template -> Click "View Config" on the menu on the top-right corner
7. **It shows nothing**

**Comment by johnliu55tw** [comment]
## Root cause
Users with the "Member" role are not granted to view `secrets` resources.

## Fix
The permissions to view the Harvester resources are controlled by ClusterRole [harvesterhci.io:view](https://github.com/harvester/harvester/blob/master/deploy/charts/harvester/templates/rbac.yaml#L53-L83). We could add resource `secrets` to this ClusterRole to grant the users with Member role to view the secrets:
```
...
rules:
- apiGroups:
    - ""
  resources:
    - secrets
  verbs:
    - get
    - list
    - watch
...
```

This ClusterRole will first be [aggregated](https://github.com/harvester/harvester/blob/9091d7d0de9e99ddc677e32c988d758065f87fbb/deploy/charts/harvester/templates/rbac.yaml#L60) into another ClusterRole `view`, then this `view` will be bound to a Group `system:authenticated` by a RoleBinding `harvester-public` in the namespace `harvester-public`. In other words, only secrets in `harvester-public` namespace will be revealed to the users with Member role.

Currently, there's another secret called `default-token-XXXXX` that contains the token to identify the service account. Not sure whether revealing this token to users? @guangbochen  @yasker 
```
kubectl get secrets -n harvester-public
NAME                  TYPE                                  DATA   AGE
default-token-58t64   kubernetes.io/service-account-token   3      30m  <- The token secret
test-public-qx7ar     secret                                1      11m
```

**Comment by TachunLin** [comment]
Verified **fixed** on master-bfd6a8a9-head. (02/21) Close this issue.

### Result
Standard user with member role in harvester can view template config and access secret.  
![image](https://user-images.githubusercontent.com/29251855/154899531-f364743b-3e29-4bf6-8e6b-2f32a797793e.png)

### Test Information
* Test Environment: 1 node1 harvester on local kvm machine
* Harvester version: master-bfd6a8a9-head
* Rancher version: v2.6.4-rc4

### Verify Steps
1. Import existing harvester from rancher dashboard
2. Create a standard user `someuser` in Rancher user & authentication
3. Access harvester from rancher virtual management 
4. Add a new member in the RBAC in harvester dashboard
5. Provide `someuser` with member role
6. Logout and login `someone` in rancher
7. Access Harvester from rancher dashboard
8. Go to Advanced/Templates -> Click any template -> Click "View Config"

### Additional Context
![image](https://user-images.githubusercontent.com/29251855/154898692-3d1d8854-3396-49bf-a636-f0b6003aca90.png)
